resources/views/delete_comment.blade.php

```diff
@@ -0,0 +1,48 @@
+<!DOCTYPE html>
+<html>
+    <head>
+        <meta charset="utf-8">
+        <meta name="viewport" content="width=device-width, initial-scale=1">
+
+        <title> 新增評論 </title>
+
+        <!-- Fonts -->
+        <link href="https://fonts.googleapis.com/css2?family=Nunito:wght@200;600&display=swap" rel="stylesheet">
+
+        <style>
+
+        </style>
+    </head>
+    <body>
+        <div><form action="/comments/delete" method="post">
+            {{ csrf_field() }}
+            <?php
+                $isCommentsValid = count($comments);
+                if ($isCommentsValid) {
+                    echo "<label for='select_id'> 選取刪除評論 </label>";
+                    echo "<select name='select_id'>";
+                    foreach ($comments as $comment) {
+                        $id = $comment->id;
+                        $student = $comment->student->getFullNameAttribute();
+                        $course = $comment->course->name;
+                        $score = $comment->score;
+                        $comm = $comment->comment;
+
+                        $str = "評論者：".$student." 課程：".$course." 分數：".$score." 內容：".$comm;
+
+                        echo "<option value=$id> $str </option>";
+                    }
+                    echo "</select>";
+                    echo "<p>";
+                } else {
+                    echo "<h1>Error: No Comment Exist</h1>";
+                }
+            ?>
```

Comment:
這邊要注意防止 XSS 
* 一種是加上 htmlspecialchars
* 一種藉由 blade 處理進行顯示
```suggestion
            <?php
                $isCommentsValid = count($comments);
                if ($isCommentsValid) {
             ?>
             <label for='select_id'> 選取刪除評論 </label>
                 <select name='select_id'>
                 <?php
                    foreach ($comments as $comment) {
                        $id = $comment->id;
                        $student = $comment->student->getFullNameAttribute();
                        $course = $comment->course->name;
                        $score = $comment->score;
                        $comm = $comment->comment;
                        $str = "評論者：".$student." 課程：".$course." 分數：".$score." 內容：".$comm;
                        ?>
                        <option value="{{ $id }}">{{ $str }}</option>
                    <?php
                    }
                     ?>
                    </select>
               <?php
                } else {
                    echo "<h1>Error: No Comment Exist</h1>";
                }
            ?>
```

Reply:
已應用